packages/mongodb-chatbot-server/src/routes/generateResponse.test.ts

```diff
@@ -231,6 +231,36 @@ describe("generateResponse", () => {
     const data = res._getData();
     expect(data).toBe("");
   });
+  it("should stream metadata", async () => {
+    const metadata = { foo: "bar", baz: 42 };
+    const staticResponse = {
+      role: "assistant",
+      content: "static response",
+      metadata,
+    } satisfies AssistantMessage;
+
+    await generateResponse({
+      ...baseArgs,
+      shouldStream: true,
+      async generateUserPrompt() {
+        return {
+          userMessage: {
+            role: "user",
+            content: "test metadata",
+          },
+          staticResponse,
+        };
+      },
+    });
+
+    const data = res._getData();
+
+    // The metadata should be streamed before the assistant response
```

Comment:
nit - I'm not sure that this is necessarily a guarantee.

For example, we in theory support the following:
1. half of the assistant message gets streamed in
2. A `metadata` event arrives that indicates the message is verified (or some other future metadata-driven state)
3. The UI updates to reflect the latest metadata
4. The rest of the assistant message streams in